docs/src/js/base/main.js

```diff
@@ -342,3 +342,65 @@ document.querySelectorAll('.switch:not(.switch--form)').forEach(function(widgetE
         console.log(e.type, e.detail);
     });
 });
+
+function getDefaultToastOptions() {
```

Comment:
Not sure if this part needs to be so complex. 
What I did for `drawer` is just copied the dialog one and changed the classes as needed. Instead of window just label it as the outer. 
I would start with the `dialog` one and go from there. 
In case you don't want it to close on document press, you would just remove that `dialog.addEventListener('click', handleClose, true);` which then you can add the close button to close the modal and call the `handleClose`
https://github.com/eBay/skin/blob/master/docs/src/js/base/main.js#L66-L116

Reply:
I just added a single method to accommodate transition and default variants. The options are just so that the callers need not pass all properties. Let me know if this is complex enough for a change :)

Reply:
I mean it isn’t really needed to be that complex. Remember we have just minimal js to make the examples work. But we don’t need to go for a full on implementation with different options and such
Again take a look at the way we did dialog (and drawer/bottom container). I would try to follow that as closely as possibly.